scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -2480,8 +2630,14 @@ def test_zero_rotation_get_set(xp):
     r_slice = r[:0]
     assert len(r_slice) == 0
 
-    with pytest.raises(IndexError):
-        r[xp.asarray([0])]
+    # TODO: Should we manually check and raise for jax here? Jax does not raise an
+    # IndexError, but a TypeError.
```

Comment:
Please open ticket upstream.
As a rule of thumb, please don't work around upstream bugs in scipy.

Reply:
Is this an error though, or does the API leave the type of error raised here unspecified? If so, `jax` can raise a `TypeError`, the question is just if we want to propagate a `TypeError` or raise an `IndexError` from it for consistency.

Reply:
I.e. 
```python
import jax.numpy as jnp

x = jnp.zeros((0, 1))
x[jnp.asarray([0])]  # TypeError. Does the spec **require** an IndexError?
```

Reply:
Mmmh. JAX diverges from numpy for out-of-bounds indexing, and the behaviour below _will_ change when you are inside the JIT. The Array API explicitly states that bounds checking is out of scope https://data-apis.org/array-api/latest/API_specification/indexing.html#single-axis-indexing.
IMHO I would skip the check altogether.